paddle/fluid/imperative/prepared_operator.cc

```diff
@@ -91,6 +85,13 @@ PreparedOp PrepareOpImpl(const NameVarMap<VarType>& ins,
   auto& kernels = kernels_iter->second;
 
   framework::RuntimeContext ctx({}, {});
+#ifdef PADDLE_WITH_MKLDNN
+  // MKLDNN variant of code reads attributes in some of GetKernelTypeForVar and
+  // GetKernelType functions, so we need to copy the attributes there.
+  // Const qualifier of Attrs had to be discarded to overwrite it.
+  auto& mutable_op_attrs = const_cast<framework::AttributeMap&>(op.Attrs());
+  mutable_op_attrs = attrs;
```

Comment:
变量mutable_op_attrs 没有在下面被使用，这个是为了后续的工作优化，还是有什么特殊的机制。
从测试来看，这里发生了一次framework::AttributeMap 的copy，framework::AttributeMap 是一个不那么高效的数据结构，使得动态图的一个任务ptb lm性能下降了5%，看下这个地方有没有策略可以进行优化

Reply:
@sfraczek @lidanqing-intel @jczaja Could we remove line 92-93, since it makes[ ptb lm](https://github.com/PaddlePaddle/benchmark/tree/master/dynamic_graph/ptb) performance decrease 5%.